GRDB/Record/FetchableRecord+Decodable.swift

```diff
@@ -79,7 +79,12 @@ private struct RowKeyedDecodingContainer<Key: CodingKey>: KeyedDecodingContainer
             } else if dbValue.isNull {
                 return nil
             } else {
-                return try T(from: RowDecoder(row: row, codingPath: codingPath + [key]))
+                if let data = row.dataNoCopy(named: key.stringValue), let dataString = String(data: data, encoding: .utf8), JSONSerialization.isValidJSONObject([dataString]) {
```

Comment:
On top of my request for further information above, I have several questions here:

- Why do you read data here, and save a string below?
- Why isn't this code also present in `decode<T>(_: forKey:)`? It looks like an oversight. Caring about your use case (decoding an optional Array) is legit, but I'd like that the library remains as consistent and welcoming as it is today. This means thinking about other people as well, including those who, for example, decode a non-optional Array.
- What about the `JSONSerialization.isValidJSONObject` performance hit? Is it really useful?
- I'm not sure `JSONSerialization.isValidJSONObject([dataString])` really checks what you expect it to check. To me it looks like it always return true.
- That `if` prevents types from providing customized decoding of JSON data. I'd like to read an exploration of the potential harm, if any.